cyclos_configuration/scripts/direct_debits/Operation_UserRecordManager.groovy

```diff
@@ -0,0 +1,21 @@
+import org.cyclos.entities.users.UserRecord
+
+/**
+  * Custom operation script with Scope 'Record'. Enables financial admins to manage directDebit user records.
+  * Financial admins can only apply specific changes, for example to mark a record as failed if the bank 
+  * informs us that an incasso has failed. Each of these possible changes has its own custom operation, but they 
+  * all share this same script. The 'action' scriptparameter of the custom operation indicates which situation is applicable.
+  */
+
+Map<String, Object> formParameters = binding.formParameters
+Map<String, String> scriptParameters = binding.scriptParameters
+UserRecord record = binding.record
+String action = scriptParameters.action
+
+new DirectDebits(binding).updateDirectDebit(record, action, formParameters)
+
+return [
+    notification: "De incasso is succesvol gewijzigd.",
+    backToRoot: true,
```

Comment:
Is it normal that backToRoot only works if the custom operation has at least one form field?

Reply:
This is a very complicated subject, with many cases and branches, implemented 3 times (classic frontend, new frontend and mobile app).
We have some open issues with this, but maybe if you can point us the expected behavior and the current behavior, we can find and fix a bug.

Reply:
Possibly, we already are 'on the root page' if the internal operation does not have any form fields. The root page then is the Record view/edit screen. So I suppose it works according to specs indeed.